plugins/solidigm/solidigm-log-page-dir.c

```diff
@@ -0,0 +1,292 @@
+
+	// Standard logs are less than 0xC0
+	if (lid < 0xC0)
+		return lid_desc_from_struct(lid, standard_log_size, standard_log_descs);
+	else if (uuid == OCP_UUID)
+		return lid_desc_from_struct(lid, ocp_log_size, ocp_log_descs);
+	// Otherwise these are Solidigm logs.
+	else if (uuid == SOLIDIGM_UUID)
+		return lid_desc_from_struct(lid, solidigm_log_size, solidigm_log_descs);
+	else if (uuid == NO_UUID)
+		return lid_desc_from_struct(lid, all_vu_log_size, all_vu_log_descs);
+
+	return "Unknown";
+}
+
+static void solidigm_supported_log_pages_print(const struct supported_log_pages *supported,
+					       const enum solidigm_uuid uuid)
+{
+	printf("Log Page Directory Log:\n");
+	printf("  Supported:\n");
+
+	for (int lid = 0; lid < sizeof(supported->supported) / sizeof(__u32); lid++) {
+		if (supported->supported[lid] == 0)
+			continue;
+
+		printf("    Log Page:\n");
+		printf("      %-16s0x%02x\n", "LID:", le32_to_cpu(lid));
+		printf("      %-16s%s\n", "Description:", lid_to_desc(lid, uuid));
+	}
+
+	printf("\n");
+}
+
+static void solidigm_supported_log_pages_json(const struct supported_log_pages *supported,
+					      const enum solidigm_uuid uuid)
+{
+	struct json_object *root = json_create_object();
+	struct json_object *supported_arry = json_create_array();
+
+	for (int lid = 0; lid < sizeof(supported->supported) / sizeof(__u32); lid++) {
+		if (supported->supported[lid] == 0)
+			continue;
+
+		struct json_object *supported_obj = json_create_object();
+
+		json_object_add_value_uint(supported_obj, "lid", le32_to_cpu(lid));
+		json_object_add_value_string(supported_obj, "description",
+					     lid_to_desc(lid, uuid));
+
+		json_array_add_value_object(supported_arry, supported_obj);
+	}
+
+	json_object_add_value_array(root, "supported", supported_arry);
+
+	json_print_object(root, NULL);
+	json_free_object(root);
+
+	printf("\n");
+}
+
+int solidigm_get_log_page_directory_log(int argc, char **argv, struct command *cmd,
+					struct plugin *plugin)
+{
+	const __u8 log_id = 0x00;
+	int uuid_index = 0;
+	enum solidigm_uuid uuid = INVALID_UUID;
+
+	const char *description = "Retrieves and parses supported log pages log.";
+	char *format = "normal";
+
+	OPT_ARGS(options) = {
+		OPT_INT("uuid-index", 'u', &uuid_index, "UUID index value : (integer)"),
+		OPT_FMT("output-format", 'o', &format, "output format : normal | json"),
+		OPT_END()
+	};
+
+	struct nvme_dev *dev = NULL;
+	int err = parse_and_open(&dev, argc, argv, description, options);
+
+	if (err)
+		return err;
+
+	struct supported_log_pages supported_data = { 0 };
+
+	struct nvme_get_log_args args = {
+		.lpo = 0,
+		.result = NULL,
+		.log = &supported_data,
+		.args_size = sizeof(args),
+		.fd = dev_fd(dev),
+		.timeout = NVME_DEFAULT_IOCTL_TIMEOUT,
+		.lid = log_id,
+		.len = sizeof(supported_data),
+		.nsid = NVME_NSID_ALL,
+		.csi = NVME_CSI_NVM,
+		.lsi = NVME_LOG_LSI_NONE,
+		.lsp = 0,
+		.uuidx = uuid_index,
+		.rae = false,
+		.ot = false,
+	};
+
+	err = nvme_get_log(&args);
```

Comment:
Maybe in the future extend nvme_cli_get_log_supported_log_pages() to accept UUID index as a parameter,

Reply:
Updating to just iterate across all valid UUIDs

Reply:
I'm doing this. Will have a new commit up today